### PR TITLE
feat: handle guest 1:1 conversation when legal hold requested [WPB-5937]

### PR DESCRIPTION
**Pull request by saleniuk**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-5937" title="WPB-5937" target="_blank">StoryWPB-5937</a>  [Android] Handling already connected guests in 1:1 conversations
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
    - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
    - [x] contains a reference JIRA issue number like `SQPIT-764`
    - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
    - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

When a user becomes under legal hold then their 1:1 conversations with guests should be disabled due to one side not having legal hold consent.

### Solutions

Check if the 1:1 conversation's legal hold status is `ENABLED` or `DEGRADED` (it means that one of members got the legal hold request and another hasn't consent to legal hold). If so, then disable messsage composer and all retry failed message buttons.
Also, fixed message composer previews.

### Dependencies (Optional)

Needs releases with:

- https://github.com/wireapp/kalium/pull/2580

#### Test Coverage (Optional)

- [x] I have added automated test to this contribution

#### How to Test

Wile being logged in as a guest, have a 1:1 conversation with some team member and request legal hold for that team member - the conversation should be degraded and sending messages should be disabled.

### Attachments (Optional)

https://github.com/wireapp/wire-android/assets/30429749/fed1c743-eccf-4354-9fe4-2467f594351b

----
#### PR Post Submission Checklist for internal contributors (Optional)

- [x] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

- [x] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [3470](https://10.10.124.134/job/AR-build-pipeline/3470/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.7.0(9060594)(9060594)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.7.0(9060594)(9060594)-staging-compat.apk)

File: app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -1355,6 +1355,7 @@
     <string name="legal_hold_system_message_disabled_others">Legal hold deactivated for %1$s.</string>
     <string name="legal_hold_system_message_disabled_conversation">Legal hold is no longer active for this conversation.</string>
     <string name="legal_hold_system_message_enabled_conversation">Legal hold is now active for this conversation.</string>
+    <string name="legal_hold_system_message_interaction_disabled">You can’t write messages with this user, as they are not part of your team and one of you is subject to legal hold. Once legal hold is deactivated, you can write messages again.</string>
```
Comment (alexandreferris) [inline app/src/main/res/values/strings.xml:1360]: shouldn't it be `write messages to this user` or maybe use `exchange messages with this user` 🤔

**Review by alexandreferris — APPROVED** [review]
Looking great! I just have one tiny nitpick, more of a question if the writing makes sense

**Review by ohassine — APPROVED** [review]

File: app/src/main/res/values/strings.xml (hunk 1 above)
Comment (saleniuk) [reply app/src/main/res/values/strings.xml:1360]: @alexandreferris  I copied this text directly from the designs so it should be ok 🙃 
image

**Comment by AndroidBob** [comment]
Build [3553](https://10.10.124.134/job/AR-build-pipeline/3553/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.7.0(9129586)(9129586)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.7.0(9129586)(9129586)-staging-compat.apk)

**Review by yamilmedina — APPROVED** [review]

**Comment by AndroidBob** [comment]
Build [3567](https://10.10.124.134/job/AR-build-pipeline/3567/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.7.0(9130193)(9130193)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.7.0(9130193)(9130193)-staging-compat.apk)

**Pull request merged**